silkworm/db/kv/api/local_transaction.cpp

```diff
@@ -66,11 +68,16 @@ Task<std::shared_ptr<CursorDupSort>> LocalTransaction::get_cursor(const std::str
 
 std::shared_ptr<chain::ChainStorage> LocalTransaction::create_storage() {
     // The calling thread *must* be the *same* which created this LocalTransaction instance
-    return std::make_shared<chain::LocalChainStorage>(DataModel{txn_, data_store_.blocks_repository});
+    return std::make_shared<chain::LocalChainStorage>(DataModel{tx_, data_store_.blocks_repository});
 }
 
-Task<TxnId> LocalTransaction::first_txn_num_in_block(BlockNum /*block_num*/) {
-    throw std::logic_error{"LocalTransaction::first_txn_num_in_block: not yet implemented"};
+Task<TxnId> LocalTransaction::first_txn_num_in_block(BlockNum block_num) {
+    auto canonical_body_for_storage = [this](BlockNum bn) -> Task<std::optional<Bytes>> {
+        DataModel access_layer{tx_, data_store_.blocks_repository};
+        co_return access_layer.read_raw_body_for_storage_from_snapshot(bn);
```

Comment:
there's this extra functionality in this PR not related to logs, did you intend to update the PR description or split a new PR?

Reply:
It was in a separate PR #2713 based on this one, then @JacekGlen merged the former into the latter